lambeq/rewrite/base.py

```diff
@@ -393,3 +399,16 @@ def _ar(self, box: Box) -> Diagram:
 
     def _ob(self, ob: Ty) -> Ty:
         return ob
+
+
+class ImperativeRewriteRule():
+    """A rewrite rule for imperative sentence diagrams.
+    """
+    def matches(self, diagram: Diagram) -> bool:
+        return diagram.cod == N.r @ S
```

Comment:
This deviates from your description that states that it handles diagrams with multiple open wires.